tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -85,19 +86,63 @@ ROOT::Experimental::RNTupleImporter::Create(std::string_view sourceFileName, std
                                             std::string_view destFileName)
 {
    auto importer = std::unique_ptr<RNTupleImporter>(new RNTupleImporter());
-   importer->fNTupleName = treeName;
+   importer->fNTupleNames[treeName] = treeName;
    importer->fSourceFile = std::unique_ptr<TFile>(TFile::Open(std::string(sourceFileName).c_str()));
    if (!importer->fSourceFile || importer->fSourceFile->IsZombie()) {
       return R__FAIL("cannot open source file " + std::string(sourceFileName));
    }
 
-   importer->fSourceTree = importer->fSourceFile->Get<TTree>(std::string(treeName).c_str());
-   if (!importer->fSourceTree) {
+   auto sourceTree = importer->fSourceFile->Get<TTree>(std::string(treeName).c_str());
+   if (!sourceTree) {
       return R__FAIL("cannot read TTree " + std::string(treeName) + " from " + std::string(sourceFileName));
    }
-
    // If we have IMT enabled, its best use is for parallel page compression
-   importer->fSourceTree->SetImplicitMT(false);
+   sourceTree->SetImplicitMT(false);
+   importer->fSourceTrees.emplace_back(sourceTree);
+
+   auto result = importer->InitDestination(destFileName);
+
+   if (!result)
+      return R__FORWARD_ERROR(result);
+
+   return importer;
+}
+
+ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::RNTupleImporter>>
+ROOT::Experimental::RNTupleImporter::Create(std::string_view sourceFileName, std::string_view destFileName)
+{
+   auto importer = std::unique_ptr<RNTupleImporter>(new RNTupleImporter());
+   importer->fSourceFile = std::unique_ptr<TFile>(TFile::Open(std::string(sourceFileName).c_str()));
+   if (!importer->fSourceFile || importer->fSourceFile->IsZombie()) {
+      return R__FAIL("cannot open source file " + std::string(sourceFileName));
+   }
+
+   auto objLink = importer->fSourceFile->GetListOfKeys()->FirstLink();
+   while (objLink) {
+      auto key = (TKey *)objLink->GetObject();
+      auto keyName = key->GetName();
+      // NOTE: Do we also want to support TNTuples?
+      bool isTree = strcmp(key->GetClassName(), "TTree") == 0;
+      // The actual current cycle number doesn't matter since TFile::Get will automatically get it from the tree name.
+      // We just need to make sure each tree is only added once.
+      bool isFirstOccurrence = key->GetCycle() == 1;
```

Comment:
It may be simpler just to check if the name is already use in `importer->fNTupleNames`